src/render/SDL_render.c

```diff
@@ -2993,6 +2941,7 @@ bool SDL_SetRenderViewport(SDL_Renderer *renderer, const SDL_Rect *rect)
     CHECK_RENDERER_MAGIC(renderer, false);
 
     if (rect) {
+        // !!! FIXME: fail if rect->w or rect->h are negative (we use this to mean "the whole viewport").
```

Comment:
Again

Reply:
Fixed in a77da699680a549b701e1c6f7d31fe22754af852.